scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,374 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+import org.apache.commons.net.ftp.FTP;
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.commons.net.ftp.FTPFile;
+import org.apache.commons.net.ftp.FTPReply;
+import org.ow2.proactive.scheduler.common.task.TaskResult;
+import org.ow2.proactive.scheduler.common.task.executable.JavaExecutable;
+
+
+/**
+ * Import and export file(s) using FTPConnector server.
+ * This task can be launched with parameters.
+ *
+ * @author The ProActive Team
+ */
+public class FTPConnector extends JavaExecutable {
+
+    private static final String DEFAULT_FTP_HOSTNAME = "localhost";
+
+    private static final int DEFAULT_FTP_PORT = 21;
+
+    private String ftpHostname = DEFAULT_FTP_HOSTNAME;
+
+    private int ftpPort = DEFAULT_FTP_PORT;
+
+    private boolean ftpExtractArchive;
+
+    private String ftpLocalFilePath = null;
+
+    private String ftpRemoteFilePath = null;
+
+    private String ftpMode = null;
+
+    private String ftpUsername = null;
+
+    private String ftpPassword = null;
+
+    /**
+     * @see JavaExecutable#init(Map)
+     */
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+        if (args.containsKey("ftpHostname")) {
+            ftpHostname = args.get("ftpHostname").toString();
+        }
+        if (args.containsKey("ftpPort")) {
+            try {
+                ftpPort = Integer.parseInt(args.get("ftpPort").toString());
+            } catch (NumberFormatException e) {
+            }
+        }
+        if (args.containsKey("ftpExtractArchive")) {
+            ftpExtractArchive = Boolean.parseBoolean(args.get("ftpExtractArchive").toString());
+        }
+        if (args.containsKey("ftpLocalFilePath")) {
+            ftpLocalFilePath = args.get("ftpLocalFilePath") == null ? "" : args.get("ftpLocalFilePath").toString();
+        }
+        if (args.containsKey("ftpRemoteFilePath")) {
+            ftpRemoteFilePath = args.get("ftpRemoteFilePath").toString();
+        }
+        if (args.containsKey("ftpMode")) {
+            ftpMode = args.get("ftpMode").toString();
+        }
+        ftpUsername = getThirdPartyCredential("FTP_USERNAME");
+        ftpPassword = getThirdPartyCredential("FTP_PASSWORD");
+        if (ftpUsername == null || ftpPassword == null) {
+            throw new IllegalArgumentException("You first need to add your ftp username and password (FTP_USERNAME, FTP_PASSWORD) to the third party credentials");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException, IllegalArgumentException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 56 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

Reply:
Agree

Reply:
done